packages/asciidoctor/index.d.ts

```diff
@@ -0,0 +1,19 @@
+export interface IRuntime {
+  ioModule: string | 'node';
+  platform: string | 'node';
+  engine: string | 'v8';
+  framework: string;
+}
+
+export class Asciidoctor {
+  getVersion(): string;
+  getCoreVersion(): string;
+  getRuntime(): IRuntime;
+  convert(input: string | Buffer, options?: any): string | Document;
```

Comment:
I think we need to define `Document` right?

Reply:
As I understand it, this is a standard document from the dom tree.  Typescript has its basic support

Reply:
![document](https://user-images.githubusercontent.com/10329528/65495057-bac2ec80-debe-11e9-93fc-b2e4641f83f3.png)

Reply:
Document is an Asciidoctor Document object. It holds the parsed blocks. That object is returned if to_file is a writable File object. Instead of returning undefined, we return the parsed document for further inspection.

Reply:
Got it, thanks

Reply:
Asciidoctor does not necessarily convert an AsciiDoc content to an HTML page. 
In this case the `Document` object is an AST: https://github.com/asciidoctor/asciidoctor/blob/master/lib/asciidoctor/document.rb that represents the structure of the AsciiDoc.

Here's the link to the API documentation: 

http://asciidoctor.github.io/asciidoctor.js/master/#asciidoctorconvert
http://asciidoctor.github.io/asciidoctor.js/master/#document

Reply:
Dan was faster :sweat_smile:

Reply:
Thanks for the links to documentation, this is very useful for me!

Reply:
You're welcome!
I realize that mapping the entire API at once might be a tedious task. So if you don't want to spend too much time on it that's OK :wink: 
I believe we can iterate on the definition file to keep improving the Typescript support, right?

Reply:
Right, we can add declarations iteratively